src/crimson/os/seastore/random_block_manager/nvme_block_device.h

```diff
@@ -253,55 +253,24 @@ class NVMeBlockDevice : public RBMDevice {
     uint64_t offset, size_t len, void *buffer_ptr);
 
   stat_device_ret stat_device() final {
-    return seastar::file_stat(device_path, seastar::follow_symlink::yes
+    auto stat = co_await seastar::file_stat(device_path, seastar::follow_symlink::yes
     ).handle_exception([](auto e) -> stat_device_ret {
       return crimson::ct_error::input_output_error::make();
-    }).then([this](auto stat) {
-      return seastar::open_file_dma(
-	device_path,
-	seastar::open_flags::rw | seastar::open_flags::dsync
-      ).then([this, stat](auto file) mutable {
-	return seastar::do_with(
-	  file,
-	  [this, stat](auto &file) mutable 
-	{
-	  return file.size().then([this, stat, &file](auto size) mutable {
-	    stat.size = size;
-	    return stat_device_ret(
-	      read_ertr::ready_future_marker{},
-	      stat
-	    );
-	    return identify_namespace(file
-	    ).safe_then([stat] (auto id_namespace_data) mutable {
-	      // LBA format provides LBA size which is power of 2. LBA is the
-	      // minimum size of read and write.
-	      stat.block_size = (1 << id_namespace_data.lbaf[0].lbads);
-	      if (stat.block_size < RBM_SUPERBLOCK_SIZE) {
-		stat.block_size = RBM_SUPERBLOCK_SIZE;
-	      } 
-	      return stat_device_ret(
-		read_ertr::ready_future_marker{},
-		stat
-	      );
-	    }).handle_error(crimson::ct_error::input_output_error::handle(
-	      [stat]{
-	      return stat_device_ret(
-		read_ertr::ready_future_marker{},
-		stat
-	      );
-	    }), crimson::ct_error::pass_further_all{});
-	  }).safe_then([&file](auto st) mutable {
-	    return file.close(
-	    ).then([st] {
-	      return stat_device_ret(
-		read_ertr::ready_future_marker{},
-		st
-	      );
-	    });
-	  });
-	});
-      });
     });
+    auto file = co_await seastar::open_file_dma(
+      device_path,
+      seastar::open_flags::rw | seastar::open_flags::dsync);
+    auto size = co_await file.size();
+    stat.size = size;
+    auto id_namespace_data = co_await identify_namespace(file);
```

Comment:
Seems like `identify_namespace` should return std::optional instead of an errorator since we use it to query available features.
Then we could:
```
auto id_namespace_data = co_await identify_namespace(file);
if (id_namespace_data) {
  ..
```
In the meantime we can instead move this into a try-catch block. However, since this will be a broader change than what I initially expected - we can defer this for later pr (after the bug itself is fixed).

@myoungwon, please let me know how you would prefer to handle this. I'm ok with also closing this PR for now.

Reply:
Thanks for reviewing, @Matan-B! Can we defer this to a later PR? I’ll address this based on your comment.